http/server/handler.go

```diff
@@ -37,21 +37,23 @@ func NewHandler(rsService api.RulesetService, cfg Config) http.Handler {
 	}
 
 	rulesetsHandler := rulesetAPI{
-		rulesets:     rsService,
-		timeout:      cfg.Timeout,
-		watchTimeout: cfg.WatchTimeout,
+		rulesets: rsService,
 	}
 
 	// router
 	mux := http.NewServeMux()
-	mux.HandleFunc("/rulesets/", func(w http.ResponseWriter, r *http.Request) {
+	mux.Handle("/rulesets/", http.StripPrefix("/rulesets/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if _, ok := r.URL.Query()["watch"]; ok && r.Method == "GET" {
-			rulesetsHandler.ServeHTTP(w, r.WithContext(cfg.WatchCancelCtx))
+			ctx, cancel := context.WithTimeout(cfg.WatchCancelCtx, cfg.WatchTimeout)
```

Comment:
Why does this need to second-guess the routing? Can't the watch code itself install its own timeout?